### ISSUE TITLE
Incorrectly displayed timestamps

### ISSUE THREAD

**Issue by akater** [alphapapa/ement.el#9]
## Overview

I'm dealing with incorrectly displayed timestamps.  Imagine a frame split vertically into two windows, with the same chatroom buffer displayed in them.  Timestamps in the left one are missing the last char:

```
┌────────┬─────────┐
│a: hey  │a: hey   │
│ 12:34:5│ 12:34:56│
│        │         │
│        │         │
│        │         │
│        │         │
│        │         │
│        │         │
└────────┴─────────┘
```

Whenever the frame is split vertically into several windows, timestamps are only displayed correctly in the rightmost one.

## Reproducing

0. I ran this in tty, with dimensions 160x57.  This may or may not be relevant.

1. Run
```sh
emacs -Q \
      -L /usr/share/emacs/site-lisp/dash \
      -L /usr/share/emacs/site-lisp/ement \
      -L /usr/share/emacs/site-lisp/plz \
      -L /usr/share/emacs/site-lisp/s \
      -L /usr/share/emacs/site-lisp/ts \
      --eval "(require 'ement)" \
      --eval "(require 'ement-room-list)"
```
(modulo path to libraries)

2. Eval `(ement-connect "@user:matrix.org" "***")`.

3. Visit a chatroom buffer.

4. Ensure the frame is split vertically into two windows with the same chatroom buffer dispalyed in both of them.

5. Observe incorrectly displayed timestamps in the left window.

**Comment by alphapapa** [comment]
Yes, I saw this when testing in a TTY, but not on a graphical display.  ISTM that it might be a bug in Emacs's `window-body-width` function, if it's returning the wrong value in a TTY, but that seems unlikely, so it's probably some tiny nuance that needs to be accounted for.  It's probably explained somewhere in the Elisp manual where window parameters are described.

**Comment by alphapapa** [comment]
It looks like this is probably the issue:

```
28.3 Window Sizes
=================
   The "body width" of a window is the width of its text area, which
does not include the scroll bar, fringes, margins or a right divider.
Note that when one or both fringes are removed (by setting their width
to zero), the display engine reserves two character cells, one on each
side of the window, for displaying the continuation and truncation
glyphs, which leaves 2 columns less for text display.  (The function
‘window-max-chars-per-line’, described below, takes this peculiarity
into account.)
```

I guess we just have to add 1 to each margin when not on a graphical display.  It seems unfortunate and a bit strange that Emacs doesn't do this automatically.

**Comment by akater** [comment]
Let me take a look at how telega (Telegram client) does it, until tomorrow.

**Comment by alphapapa** [comment]
> For your consideration. If `display-graphic-p` is employed, we might not be able to display the same buffer consistently in different (GUI and TUI) frames.

Probably so.

>  OTOH, padding with `?\s` on the right might just be sufficient.

Maybe I'm misunderstanding something, but I don't think that will work, because it will just make the string longer, and the right end of it is already past the right edge of the window.  I think the margins need to be one character wider on TTYs, but I don't know the right way to do that (one of the reasons being that a buffer might be displayed on both a GUI and TTY frame, even though that's probably unlikely and not a situation we must account for).

**Comment by alphapapa** [comment]
I just noticed something:

https://github.com/alphapapa/ement.el/blob/e10f70f46dd147df57c92bd60b41e86c95747e90/ement-room.el#L1464

I don't know if the space being propertized there ends up being displayed.  Maybe if we use a `display` property on that space string to hide it, that might make a difference.

**Comment by akater** [comment]
> it will just make the string longer, and the right end
> of it is already past the right edge of the window

Right.  What I meant was, “to append `?\s` and align the whole thing to the right”.

> Maybe if we use a `display` property on that space string
> to hide it, that might make a difference.

That's beyond my expertise.  I think I offered all help on this I could offer efficiently enough.

**Comment by alphapapa** [comment]
@treed helpfully pointed out that the gaps displayed between margins and window body text are actually the fringes, and that might be relevant to this problem, since TTYs have no fringes AFAIK.

**Comment by alphapapa** [comment]
Also discussed in https://github.com/alphapapa/ement.el/issues/82.  The best solution is probably to use one of the non-margin message formats on non-graphical displays.
